docs/man/docker.1.md

```diff
@@ -292,79 +292,192 @@ inside it)
-images.
 
-#### dm.loopdatasize
-Specifies the size to use when creating the loopback file for the "data"
-device which is used for the thin pool. The default size is 100G. Note that the
-file is sparse, so it will not initially take up this much space.
+Specifies the size to use when creating the base device, which limits
+the size of images and containers. The default value is 10G. Note,
+thin devices are inherently "sparse", so a 10G device which is mostly
+empty doesn't use 10 GB of space on the pool. However, the filesystem
+will use more space for the empty case the larger the device
+is. **Warning**: This value affects the system-wide "base" empty
+filesystem that may already be initialized and inherited by pulled
+images.  Typically, a change to this value will require additional
+steps to take effect: 1) stop `docker -d`, 2) `rm -rf
+/var/lib/docker`, 3) start `docker -d`.
 
-#### dm.loopmetadatasize
-Specifies the size to use when creating the loopback file for the "metadadata"
-device which is used for the thin pool. The default size is 2G. Note that the
-file is sparse, so it will not initially take up this much space.
+Example use: ``docker -d --storage-opt dm.basesize=20G``
 
 #### dm.fs
-Specifies the filesystem type to use for the base device. The supported
-options are "ext4" and "xfs". The default is "ext4"
+
+Specifies the filesystem type to use for the base device. The
+supported options are `ext4` and `xfs`. The default is `ext4`.
+
+Example use: ``docker -d --storage-opt dm.fs=xfs``
 
 #### dm.mkfsarg
+
 Specifies extra mkfs arguments to be used when creating the base device.
 
+Example use: ``docker -d --storage-opt "dm.mkfsarg=-O ^has_journal"``
+
 #### dm.mountopt
+
 Specifies extra mount options used when mounting the thin devices.
 
+Example use: ``docker -d --storage-opt dm.mountopt=nodiscard``
+
+#### dm.use_deferred_removal
+
+Enables use of deferred device removal if `libdm` and kernel driver
+support the mechanism.
+
+Deferred device removal means that if device is busy when devices are
+being removed/deactivated, then a deferred removal is scheduled on
+device. And devices automatically goes away when last user of device
+exits.
+
+For example, when a container exits, its associated thin device is removed. If
+that devices has leaked into some other mount namespace and can't be removed,
+the container exit still succeeds and this option causes the system to schedule
+device for deferred removal. It does not wait in a loop trying to remove a busy
+device.
+
+Example use: ``docker -d --storage-opt dm.use_deferred_removal=true``
+
+#### dm.loopdatasize
+
+**Note**: This option configures devicemapper loopback, which is not for
+production use.
+
+Specifies the size to use when creating the loopback file for the
+"data" device which is used for the thin pool. The default size is
+100G. Note that the file is sparse, so it will not initially take up
```

Comment:
@jamtur01 this is the man-page, I think a slightly different formatting is used there.